mlflow/genai/optimize/base.py

```diff
@@ -0,0 +1,161 @@
+import inspect
+from typing import TYPE_CHECKING, Optional
+
+from mlflow.entities.model_registry import Prompt
+from mlflow.exceptions import MlflowException
+from mlflow.genai.evaluation.utils import (
+    _convert_eval_set_to_df,
+)
+from mlflow.genai.optimize.optimizers import _BaseOptimizer, _DSPyMIPROv2Optimizer
+from mlflow.genai.optimize.types import (
+    OBJECTIVE_FN,
+    LLMParams,
+    OptimizerConfig,
+    PromptOptimizationResult,
+)
+from mlflow.genai.scorers import Scorer
+from mlflow.tracking._model_registry.fluent import load_prompt, register_prompt
+from mlflow.utils.annotations import experimental
+
+if TYPE_CHECKING:
+    from genai.evaluation.utils import EvaluationDatasetTypes
+
+_ALGORITHMS = {"DSPy/MIPROv2": _DSPyMIPROv2Optimizer}
+
+
+@experimental
+def optimize_prompt(
+    *,
+    target_llm_params: LLMParams,
+    prompt_uri: str,
```

Comment:
Why do we enforce users to register the prompt first before using this API? I think they should be able to provide a prompt directly and we register for them as the initial version?

Reply:
Yep I agree, the learning cost is not high, while supporting Union[str, Prompt, ...] seems reasonable to me